server/auth_callout.go

```diff
@@ -18,22 +18,42 @@ import (
 	"crypto/tls"
 	"encoding/pem"
 	"fmt"
+	"strings"
 	"time"
 
 	"github.com/nats-io/jwt/v2"
 	"github.com/nats-io/nkeys"
 )
 
 const (
-	AuthCalloutSubject    = "$SYS.REQ.USER.AUTH"
-	AuthRequestSubject    = "nats-authorization-request"
-	AuthRequestXKeyHeader = "Nats-Server-Xkey"
+	AuthCalloutSubject     = "$SYS.REQ.USER.AUTH"
+	AuthRequestSubject     = "nats-authorization-request"
+	AuthRequestXKeyHeader  = "Nats-Server-Xkey"
+	AuthCalloutErrorHeader = "Auth-Callout-Error"
+	AuthCalloutServerIdTag = "Auth-Callout-Server-ID"
 )
 
+func getTagValue(uc *jwt.UserClaims, key string) string {
+	// we are expecting `tag: serverID` or `tag:serverID`
+	// note that tags lower case
+	k := fmt.Sprintf("%s:", strings.ToLower(key))
+	for _, t := range uc.Tags {
+		if strings.HasPrefix(t, k) {
+			prefix := len(key) + 1
+			if len(t) > prefix {
+				return strings.TrimSpace(t[prefix:])
+			}
```

Comment:
```suggestion
			return strings.TrimSpace(t[len(k):])
```

nit - this length check is redundant, [CutPrefix from go 1.20](https://cs.opensource.google/go/go/+/refs/tags/go1.20:src/strings/strings.go;l=1273-1278) doesn't use one